test/test_segment.py

```diff
@@ -0,0 +1,214 @@
+                                    segment_image
+                                )
+import AxonDeepSeg.segment
+import AxonDeepSeg
+from config import axonmyelin_suffix, axon_suffix, myelin_suffix
+
+class TestCore(object):
+    def setup(self):
+        # Get the directory where this current file is saved
+        self.testPath = Path(__file__).resolve().parent
+        self.projectPath = self.testPath.parent
+
+        self.modelPath = (
+            self.projectPath /
+            'AxonDeepSeg' /
+            'models' /
+            'model_seg_rat_axon-myelin_sem'
+            )
+
+        self.imageFolderPath = (
+            self.testPath /
+            '__test_files__' /
+            '__test_segment_files__'
+            )
+        self.relativeImageFolderPath = Path(
+            'test/__test_files__/__test_segment_files__'
+        )
+
+        self.imagePath = self.imageFolderPath / 'image.png'
+
+        self.imageFolderPathWithPixelSize = (
+            self.testPath /
+            '__test_files__' /
+            '__test_segment_files_with_pixel_size__'
+            )
+        self.imagePathWithPixelSize = self.imageFolderPathWithPixelSize / 'image.png'
+
+        self.statsFilename = 'model_statistics_validation.json'
+
+    @classmethod
+    def teardown_class(cls):
+
+        testPath = Path(__file__).resolve().parent
+        projectPath = testPath.parent
+        imageFolderPath = testPath / '__test_files__' / '__test_segment_files__'
+
+        imageFolderPathWithPixelSize = (
+            testPath /
+            '__test_files__' /
+            '__test_segment_files_with_pixel_size__'
+            )
+
+        outputFiles = [
+            'image' + str(axon_suffix),
+            'image' + str(myelin_suffix),
+            'image' + str(axonmyelin_suffix),
+            'image.nii.gz',
+            'image_2' + str(axon_suffix),
+            'image_2' + str(myelin_suffix),
+            'image_2' + str(axonmyelin_suffix),
+            'image_2.nii.gz'
+            ]
+
+        for fileName in outputFiles:
+
+            if (imageFolderPath / fileName).exists():
+                (imageFolderPath / fileName).unlink()
+
+            if (imageFolderPathWithPixelSize / fileName).exists():
+                (imageFolderPathWithPixelSize / fileName).unlink()
+
+    # --------------segment_folders tests-------------- #
+    @pytest.mark.integration
+    def test_segment_folders_creates_expected_files(self):
+        path_model = generate_default_parameters('SEM', str(self.modelPath))
+
+        overlap_value = [48,48]
+
+        outputFiles = [
+            'image' + str(axon_suffix),
+            'image' + str(myelin_suffix),
+            'image' + str(axonmyelin_suffix)
+            ]
+
+        segment_folders(
+            path_testing_images_folder=str(self.imageFolderPath),
+            path_model=str(path_model),
+            overlap_value=overlap_value,
+            acquired_resolution=0.37,
```

Comment:
Oh! Now I see where 0.37 in the notebook was coming from.
I checked the `__test_files__/__test_segment_files__` and `__test_files__/__test_segment_files_with_pixel_size__` . They are smaller than the data_test of the SEM model and 0.37 makes sense in that case.